code/modules/economy/holopay.dm

```diff
@@ -0,0 +1,241 @@
+/obj/effect/overlay/holopay
+	name = "holographic pay stand"
+	desc = "an unregistered pay stand"
+	icon = 'icons/obj/economy.dmi'
+	icon_state = "card_scanner"
+	alpha = 150
+	anchored = TRUE
+	layer = FLY_LAYER
+	/// ID linked to the holopay
+	var/datum/weakref/card_ref
+	/// Replaces the "pay whatever" functionality with a set amount when non-zero.
+	var/force_fee = 0
+	/// Minimum forced fee for holopay stations. Registers as "pay what you want."
+	var/min_fee = 0
+	/// Maximum forced fee. It's unlikely for a user to encounter this type of money, much less pay it willingly.
+	var/max_fee = 5000
+	/// Max range at which the hologram can be projected before it deletes
+	var/max_holo_range = 4
+	/// List of logos available for customization - via font awesome 5
+	var/static/list/available_logos = list("angry", "ankh", "band-aid", "cannabis", "cat", "cocktail", "coins", "comments-dollar",
+	"cross", "cut", "donate", "dna", "flask", "glass-cheers", "glass-martini-alt", "hand-holding-usd", "heart", "heart-broken",
+	"hamburger", "hat-cowboy-side", "money-check-alt", "music", "pizza-slice", "prescription-bottle-alt", "radiation", "robot", "smile",
+	"tram", "trash")
+	/// The brand icon chosen by the user
+	var/shop_logo = "donate"
+
+/**
+ * Links the source card to the holopay. Begins checking if its in range.
+ */
+/obj/effect/overlay/holopay/proc/assign_card(turf/target, obj/item/card/id/card)
+	loc = target
+	card_ref = WEAKREF(card)
+	desc = "Pays directly into [card.registered_account.account_holder]'s bank account."
+	add_atom_colour("#77abff", FIXED_COLOUR_PRIORITY)
+	set_light(2)
+	visible_message(span_notice("A holographic pay stand appears."))
+	/// Start checking if the owner has left or died
+	RegisterSignal(card, COMSIG_MOVABLE_MOVED, .proc/check_operation)
```

Comment:
This unfortunately won't work, since the card only moves when it's removed from the owner's inventory. 

On second thought, I think what you'd need to do is track both the card and the owner. if the card moves to a tile, we stop caring about owners, and if it moves into something, we track that something's movements for the range component until the card moves out of it. Handling that alongside like, recursive contents gets mildly messy though

Reply:
Good call